components/user_settings/display/user_settings_display.tsx

```diff
@@ -35,40 +36,92 @@ function getDisplayStateFromProps(props) {
-        defaultClientLocale: PropTypes.string,
-        enableThemeSelection: PropTypes.bool,
-        configTeammateNameDisplay: PropTypes.string,
-        currentUserTimezone: PropTypes.string,
-        enableTimezone: PropTypes.bool,
-        shouldAutoUpdateTimezone: PropTypes.bool,
-        militaryTime: PropTypes.string,
-        teammateNameDisplay: PropTypes.string,
-        channelDisplayMode: PropTypes.string,
-        messageDisplay: PropTypes.string,
-        collapseDisplay: PropTypes.string,
-        linkPreviewDisplay: PropTypes.string,
-        lockTeammateNameDisplay: PropTypes.bool,
-        actions: PropTypes.shape({
-            getSupportedTimezones: PropTypes.func.isRequired,
-            autoUpdateTimezone: PropTypes.func.isRequired,
-            savePreferences: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Option = {
+    value: string;
+    radionButtonText: {
+        id: string;
+        message: string;
+        moreId?: string;
+        moreMessage?: string;
+    };
+}
 
-    constructor(props) {
+type SectionProps ={
+    section: string ;
+    display: string;
+    defaultDisplay: string;
+    value: string;
+    title: {
+        id: string;
+        message: string;
+    };
+    firstOption: Option;
+    secondOption: Option;
+    thirdOption?: Option;
+    description: {
+        id: string;
+        message: string;
+    };
+    disabled?: boolean;
+}
+
+type Props = {
+    user: UserProfile;
+    updateSection: (section: string) => void;
+    activeSection?: string;
+    closeModal?: () => void;
+    collapseModal?: () => void;
+    setRequireConfirm?: () => void;
+    setEnforceFocus?: () => void;
+    timezones: string[];
+    userTimezone: UserTimezone;
+    allowCustomThemes: boolean;
+    enableLinkPreviews: boolean;
+    defaultClientLocale: string;
+    enableThemeSelection: boolean;
+    configTeammateNameDisplay: string;
+    currentUserTimezone: string;
+    enableTimezone: boolean;
+    shouldAutoUpdateTimezone: boolean;
+    lockTeammateNameDisplay: boolean;
+
+    militaryTime: string;
+    teammateNameDisplay: string;
+    channelDisplayMode: string;
+    messageDisplay: string;
+    collapseDisplay: string;
+    linkPreviewDisplay: string;
+    actions: {
+        savePreferences: (userId: string, preferences: Array<PreferenceType>) => void;
+        getSupportedTimezones: () => void;
+        autoUpdateTimezone: (deviceTimezone: string) => void;
+    };
+}
+
+type State = {
+    isSaving: boolean;
+    militaryTime: string;
+    teammateNameDisplay: string;
+    channelDisplayMode: string;
+    messageDisplay: string;
+    collapseDisplay: string;
+    linkPreviewDisplay: string;
+    handleSubmit?: () => void;
+    serverError?: string;
+    display?: string;
+}
+
+export default class UserSettingsDisplay extends React.PureComponent<Props, State> {
+    public prevSections: {
+        theme: string; // dummy value that should never match any section name
```

Comment:
Nitpick: This comment isn't really relevant here, it's just describing that the value for `theme` shouldn't match any section name since `theme` is always the first section.

Reply:
yes, I have removed it.

Reply:
We want the comment below, not here.
The comment below describes the "magic string" of `dummySectionName`.